magpie/alembic/versions/2020-11-30_9d977cce1ef2_group_permission_priority.py

```diff
@@ -0,0 +1,45 @@
+"""
+Group permission priority.
+
+Revision ID: 9d977cce1ef2
+Revises: cfe64807c143
+Create Date: 2020-11-30 15:24:57.320156
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from alembic.context import get_context  # noqa: F401
+from sqlalchemy.dialects.postgresql.base import PGDialect
+from sqlalchemy.orm.session import sessionmaker
+
+# revision identifiers, used by Alembic.
+revision = "9d977cce1ef2"
+down_revision = "cfe64807c143"
+branch_labels = None
+depends_on = None
+
+Session = sessionmaker()
+groups = sa.table(
+    "groups",
+    sa.column("priority", sa.Integer()),
+)
+
+
+def upgrade():
+    context = get_context()
+    session = Session(bind=op.get_bind())
+
+    # two following lines avoids double "DELETE" erroneous call when deleting group due to incorrect checks
+    # https://stackoverflow.com/questions/28824401
+    context.connection.engine.dialect.supports_sane_rowcount = False
+    context.connection.engine.dialect.supports_sane_multi_rowcount = False
+
+    if isinstance(context.connection.engine.dialect, PGDialect):
+        op.add_column("groups", sa.Column("priority", sa.Integer(), nullable=False, default=0, server_default="0"))
+        stmt = groups.update().where(groups.c.priority is None).values({"priority": 0})
```

Comment:
I decided to go with code override since technically we can boot with different `MAGPIE_ANONYMOUS_GROUP` value and have the literal value change name. So whatever is stored in the db doesn't matter that much anymore. I'm having also some trouble defining an `inf` for admins. 
Maybe I should remove the migration script altogether and remove the priority db field to use explicit if against the name and return -1/0/1 ?

Reply:
Given that we agreed that we want to avoid the standard groups priority hell, I guess that we could indeed keep that door fully close and don't add this field.